blockbot/blockbot.php

```diff
@@ -26,7 +27,51 @@ function blockbot_uninstall() {
 function blockbot_init_1(App $a) {
 	$crawlerDetect = new CrawlerDetect();
 
-	if ($crawlerDetect->isCrawler()) {
-		System::httpExit(403, 'Bots are not allowed');
+	$logdata = ['agent' => $_SERVER['HTTP_USER_AGENT'], 'uri' => $_SERVER['REQUEST_URI']];
+
+	if (!$crawlerDetect->isCrawler()) {
+		logger::debug('Good user agent detected', $logdata);
+		return;
+	}
+
+	// List of strings of reported false positives
+	$agents = ['Mastodon', 'hackney', 'Faraday', 'okhttp', 'UniversalFeedParser', 'PixelFedBot', 'python-requests',
+		'WordPress', 'http.rb'];
+	foreach ($agents as $agent) {
+		if (stristr($_SERVER['HTTP_USER_AGENT'], $agent)) {
+			// The agents had been reported to https://github.com/JayBizzle/Crawler-Detect/issues/
+			logger::notice('Already reported wrong detection', $logdata);
+			return;
+		}
+	}
+
+	// List of strings of known "good" agents
+	$agents = ['diaspora-connection-tester', 'DiasporaFederation', 'Friendica', '(compatible; zot)',
+		'Micro.blog', 'GangGo', 'python/federation', 'GNU social', 'winHttp',
+		'Go-http-client', 'Mr.4x3 Powered', 'Test Certificate Info', 'WordPress.com', 'zgrab',
+		'curl/', 'StatusNet', 'OpenGraphReader/', 'Uptimebot/', 'python-opengraph-jaywink',
+		'fediverse.network crawler', 'Active_Pods_CheckBot_3.0', 'Social-Relay'];
+
+	foreach ($agents as $agent) {
+		if (stristr($_SERVER['HTTP_USER_AGENT'], $agent)) {
+			// Report every false positive here: https://github.com/JayBizzle/Crawler-Detect/issues/
+			// After report move it into the array above
```

Comment:
This definitely is a development comment, I would prefer it not to be included in the "final" PR.

Reply:
It is the "develop" branch :-) Problem is: When we don't have updated this PR very soon, we really should warn every administrator to not use it by now, since it blocks communication with Mastodon and Pleroma by now.

Reply:
Yeah, that's why I'd rather to wait a bit before merging it.

Reply:
But the problem is that the addon in the current state really is blocking communication.

Reply:
Yes, I got it. It's on develop, people will disable it. If you have a quick fix, it's okay, just remove the logging calls and the comment about reporting false positives and I will merge the PR while you keep monitoring it on your node.